packages/flame_3d/lib/src/resources/light/light.dart

```diff
@@ -0,0 +1,9 @@
+import 'package:flame_3d/components.dart';
+import 'package:flame_3d/graphics.dart';
+
+abstract class Light extends Component3D {
```

Comment:
A light should be a resource while this is a component. The `graphics` and `resources` directory should always stay flame agnostic as we can in the long term pull it out as a 3D rendering framework for others to use.

We should have light describe all the "shader values" just like how materials and such do that. And on bind it should set the uniforms on the current active shader/material somehow. I am not sure how we can best do that but the rendering pipeline should imho go:
```
- Mesh
   - Surface
      - Material
         - Texture
         - Shaders
         - Uniforms
- Light
   - Uniforms on Shaders
```

I am of course open to ideas, maybe potentially light should not even be a graphics resource (like you did here) but that we have to make sure that you still can control resource uniforms so that the light can change aspects of materials

Reply:
I see. let me think about that.

we still need a way in the end to refer to the list of lights in the `bind` method of the mesh right?

so either we store it on the graphics device, or create another parameter to bind, like a `context`. what do you think?

I will try to take a look tonight on how other engines do this too, see if it can... shed any light on us.

Reply:
@wolfenrain made an attempt based on what I learned from playing with Godot.

It is hard to be 1:1 because their architecture is very different, and they have no separation between their "components" (called Nodes) and the features (like Light).